### PR TITLE
fix(cli): make blocking async fs ops be internally async

### LINKED ISSUE #7400: various async fs ops are not actually async
Example op_fdatasync_async

https://github.com/denoland/deno/blob/11afd071aec2d5ea6eb37a934c0835d8f6b1e764/cli/ops/fs.rs#L285-L300

This ought to be calling `tokio::task::spawn_blocking`.

op_stat_async is properly implemented 
https://github.com/denoland/deno/blob/11afd071aec2d5ea6eb37a934c0835d8f6b1e764/cli/ops/fs.rs#L864-L886

### PR DESCRIPTION
**Pull request by caspervonb**
This makes the following ops internally async via tokio::fs::File calls:

- op_fstat_async
- op_ftruncate_async
- op_seek_async
- op_fdatasync_async
- op_fsync_async

Fixes #7400

### CODE REVIEW

File: cli/ops/fs.rs [hunk 1]
```diff
@@ -248,14 +248,21 @@ async fn op_seek_async(
   _zero_copy: BufVec,
 ) -> Result<Value, ErrBox> {
   let (rid, seek_from) = seek_helper(args)?;
-  // TODO(ry) This is a fake async op. We need to use poll_fn,
```
Comment (caspervonb) [inline cli/ops/fs.rs:251]: There is a trait for this as described.
Is it critical to use this (future::Poll) over just calling `seek.await` @ry ?

File: cli/ops/fs.rs [hunk 2]
```diff
@@ -248,14 +248,21 @@ async fn op_seek_async(
...
+
+  let mut resource_table = state.resource_table.borrow_mut();
+  let resource_holder = resource_table
+    .get_mut::<StreamResourceHolder>(rid)
+    .ok_or_else(ErrBox::bad_resource_id)?;
+
+  let pos = match &mut resource_holder.resource {
+    StreamResource::FsFile(Some((file, _))) => {
+      file.seek(seek_from).await.map_err(ErrBox::from)
+    }
```
Comment (bartlomieju) [inline cli/ops/fs.rs:260]: This is bad - you'll be holding borrow on the resource table until `await` finishes - ie. `BorrowMutError` will pop up when you'll try to use other resource

File: cli/ops/fs.rs [hunk 3]
```diff
@@ -1308,10 +1342,21 @@ async fn op_ftruncate_async(
...
-  })?;
+
+  let mut resource_table = state.resource_table.borrow_mut();
+  let resource_holder = resource_table
+    .get_mut::<StreamResourceHolder>(rid)
+    .ok_or_else(ErrBox::bad_resource_id)?;
+
+  match &mut resource_holder.resource {
+    StreamResource::FsFile(Some((file, _))) => {
+      file.set_len(len).await.map_err(ErrBox::from)
```
Comment (bartlomieju) [inline cli/ops/fs.rs:1353]: Ditto

File: cli/ops/fs.rs (hunk 2 above)
Comment (caspervonb) [reply cli/ops/fs.rs:260]: Ah; clone the table then?

**Comment by caspervonb** [comment]
Ops have been a moving fast lately, I need to rewrite this.

**Pull request closed without merging**